### DISCUSSION TITLE
Should we rename the Agent component Coordinator?

### DISCUSSION THREAD

**Discussion by andyk** [agentos-project/agentos#143] [General]
Renaming would reduce confusion about the difference between the role of the agent component spec file (i.e., `agentos.ini`) and the component (that we currently call "agent") that coordinates all the other components.

However, the role that component currently plays maps pretty well to the classic definition of an RL agent. I.e., it can be trained, and run. It uses trainer and policy. Etc.

I think once we support hierarchical components (e.g., via component dependencies), the name Agent might be a better fit than Coordinator.

**Comment by nickjalbert** [comment]
I can see the argument either way.  I lean slightly toward (at least trying) Coordinator because it feels like we have a fair bit going on in the concept space and not reusing a term we already use in a lot of places to denote different things might make it simpler to grok the overall picture.  

Just brainstorming, but what would happen if we renamed `agentos.ini` to something like `component_manifest.{ini, yaml}`?  Could that be a strategy to avoid potential confusion?
